src/sources/opentelemetry/http.rs

```diff
@@ -103,12 +114,69 @@ pub(crate) fn build_warp_filter(
     log_filters.or(trace_filters).unify().boxed()
 }
 
+fn enrich_events(
```

Comment:
This function is similar to [`SimpleHttpSource::enrich_events`](https://github.com/vectordotdev/vector/blob/master/src/sources/http_server.rs#L428), but it only injects headers into the event.

Reply:
It looks like the code blocks that injects the headers are identical (except for the source names). I think it would be nice to convert this into a common function and move it to a new file `src/sources/util/http.rs`.